app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,32 @@
+class SessionsController < ApplicationController
+
+  # before_action :find_user
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.find_by name: params[:user][:name]
+    if @user
+      session[:user_id] = @user.id
+      flash[:success] = "Welcome back #{@user.name}!"
+    else
+      @user = User.create(user_params)
```

Comment:
You should check to see if `create` here worked.  Check to make sure validations passed!